src/test/libcephfs/multiclient.cc

```diff
@@ -97,3 +98,82 @@ TEST(LibCephFS, MulticlientHoleEOF) {
   ceph_shutdown(ca);
   ceph_shutdown(cb);
 }
+
+static void write_func(bool *stop)
+{
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, "/"), 0);
+
+  char name[20];
+  snprintf(name, sizeof(name), "foo.%d", getpid());
+  int fd = ceph_open(cmount, name, O_CREAT|O_RDWR, 0644);
+  ASSERT_LE(0, fd);
+
+  int buf_size = 4096;
+  char *buf = (char *)malloc(buf_size);
+  if (!buf) {
+    *stop = true;
+    printf("write_func failed to allocate buffer!");
+    return;
+  }
+  memset(buf, 1, buf_size);
+
+  while (!(*stop)) {
+    int i;
+
+    // truncate the file size to 4096 will set the max_size to 4MB.
+    ASSERT_EQ(0, ceph_ftruncate(cmount, fd, 4096));
+
+    // write 4MB + extra 64KB data will make client to trigger to
+    // call check_cap() to report new size. And if MDS is revoking
+    // the Fsxrw caps and we are still holding the Fw caps and will
+    // trigger tracker#57244.
+    for (i = 0; i < 1040; i++) {
+      ASSERT_EQ(ceph_write(cmount, fd, buf, buf_size, 0), buf_size);
+    }
+  }
+
+  ceph_shutdown(cmount);
+}
+
+static void setattr_func(bool *stop)
+{
+  struct ceph_mount_info *cmount;
+  ASSERT_EQ(ceph_create(&cmount, NULL), 0);
+  ASSERT_EQ(ceph_conf_read_file(cmount, NULL), 0);
+  ASSERT_EQ(0, ceph_conf_parse_env(cmount, NULL));
+  ASSERT_EQ(ceph_mount(cmount, "/"), 0);
+
+  char name[20];
+  snprintf(name, sizeof(name), "foo.%d", getpid());
+  int fd = ceph_open(cmount, name, O_CREAT|O_RDWR, 0644);
+  ASSERT_LE(0, fd);
+
+  while (!(*stop)) {
+    // setattr will make the MDS to acquire xlock for the filelock and
+    // force to revoke caps from clients
+    struct ceph_statx stx = {.stx_size = 0};
```

Comment:
the above line is causing:
```
/build/ceph-16.2.13-474-gf3910b17/src/test/libcephfs/multiclient.cc:158:43: sorry, unimplemented: non-trivial designated initializers not supported
     struct ceph_statx stx = {.stx_size = 0};
```

for pacific builds.
According to gcc.gnu.org, this feature is available in c++20.
But pacific is built with -std=c++1z


FYI - [build console logs](https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=gigantic/71992//consoleFull)

Reply:
Why I didn't see it in this tracker ?

Reply:
Fixed it. Please review @mchangir 
Thanks.

Reply:
Seems to be causing failures again: https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=gigantic/73151//consoleFull

```
/build/ceph-16.2.14-100-gb009ec3d/src/test/libcephfs/multiclient.cc: In function 'void setattr_func(bool*)':
/build/ceph-16.2.14-100-gb009ec3d/src/test/libcephfs/multiclient.cc:158:43: sorry, unimplemented: non-trivial designated initializers not supported
     struct ceph_statx stx = {.stx_size = 0};
```

@lxbsz

Reply:
Done. I have fixed it before, not sure why it back again :-(